include/dgl/array.h

```diff
@@ -285,6 +315,150 @@ IdArray VecToIdArray(const std::vector<T>& vec,
   return ret.CopyTo(ctx);
 }
 
+template<typename T>
+NDArray VecToNDArray(const std::vector<T> &vec, DLDataType dtype, DLContext ctx) {
+  int64_t size = static_cast<int64_t>(vec.size());
+  NDArray ret = NDArray::Empty({size}, dtype, DLContext{kDLCPU, 0});
+  std::copy(vec.begin(), vec.end(), static_cast<T *>(ret->data));
+  return ret.CopyTo(ctx);
+}
+
+///////////////////////// Dispatchers //////////////////////////
+
+#define ATEN_XPU_SWITCH(val, XPU, ...) do {                     \
```

Comment:
These dispatch macros look exactly like [pytorch aten dispatcher](https://github.com/pytorch/pytorch/blob/985fd970aad85799a568b76c2a1b75a1c422fe53/aten/src/ATen/Dispatch.h). Is there any possibility we depend on ATen in the future?

Reply:
Does ATen work with MXNet?